library/channel.c

```diff
@@ -513,9 +518,20 @@ static void latency_reply_cb(void *ctx, message *reply) {
         message_get_uint64_header(reply, LatencyProbeTime, &ts)) {
         ch->latency = uv_now(ch->loop) - ts;
         ZITI_LOG(VERBOSE, "ch[%d](%s) latency is now %ld", ch->id, ch->name, ch->latency);
-    } else {
+    }
+    else {
         ZITI_LOG(WARN, "invalid latency probe result ct[%d]", reply->header.content);
     }
+    uv_timer_start(&ch->timer, send_latency_probe, LATENCY_INTERVAL, 0);
+}
+
+static void latency_timeout(uv_timer_t *t) {
+    ziti_channel_t *ch = t->data;
+    ziti_channel_remove_waiter(ch, ch->latency_waiter);
+    ch->latency_waiter = NULL;
```

Comment:
noob question, why not NULL in ziti_channel_remove_waiter? bad form? seems like you could put it close to the waiter free?

Reply:
it won't work, pass-by-value